vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
Every call to connection.cursor resets the connection already and I think within a cursor, we don't have to reset_connection. If a legitimate scenario needs reset connection, I think it should use a different cursor. Ideally, from my experience, logical set of queries should use individual cursor each. Let me know if there is a different scenario that you know of needing a reset_connection.